Feat(eos_cli_config_gen): Add support for monitor loop-protection

dmilyt opened the pull request:
## Change Summary

<!-- Enter short PR description -->
Add support for monitor loop-protection.
Create a new data model and templates to render the following configuration:
```
monitor loop-protection
   no schutdown
   disabled-time 100
   protect vlan 1000-1100
   rate-limit 100
   transmit-interval 10 
```
And updated ethernet_interfaces data model and templates to render following configuration:
```
interface Ethernet 1
   no loop-protection
```
  
## Related Issue(s)

Fixes #6012

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
<!--- Describe your changes in detail -->
Add the support for monitor loop protection.
```
monitor_loop_protection:
  enabled: <bool>
  disabled_time: <int, 0-604800; default 604800>
  protect_vlan: <str>
  rate_limit: <int; <0-1000>
  transmit_interval: <int; 1-10; default 5>
```
Update schema and data model under <b>ethernet_interfaces:</b>
```
ethernet_interfaces:
  - name: <str; required; unique>
    <snip>
    loop_protection: <bool>
    <snip>
```

<!--- Describe data model implemented for new features -->

## How to test
<!--- Please describe in detail how you tested your changes. -->
molecule test
<!--- Include details of your testing environment, and the tests you ran to -->

## Checklist

### User Checklist

<!-- Add your own checklist using MD syntax and by replacing N/A -->
- N/A

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.arista.com/stable/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/extensions/molecule) testing accordingly. (check the box if not applicable)

dmilyt committed:
schema files

dmilyt committed:
molecule files

dmilyt committed:
Feat(eos_cli_config_gen): Add support for monitor loop-protection

dmilyt committed:
fix conflicts and generate molecule tests

dmilyt committed:
Fix linting in documentation

dmilyt committed:
Merge branch 'devel' into 6012

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml:
```diff
@@ -62,6 +62,9 @@ keys:
             "l2_mru" should only be defined for platforms supporting the "l2 mru" CLI.
+        loop_protection:
+          type: bool
+          description: Enable loop protection.
```
```suggestion
          description: Enable/disable loop protection.
```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,27 @@
+{%     endif %}
+{%     if monitor_loop_protection.transmit_interval is arista.avd.defined %}
+   transmit-interval {{ monitor_loop_protection.transmit_interval }}
+{%     endif %}
```
```suggestion
{%     if monitor_loop_protection.protect_vlan is arista.avd.defined %}
   protect vlan {{ monitor_loop_protection.protect_vlan }}
{%     endif %}
{%     if monitor_loop_protection.rate_limit is arista.avd.defined %}
   rate-limit {{ monitor_loop_protection.rate_limit }}
{%     endif %}
{%     if monitor_loop_protection.transmit_interval is arista.avd.defined %}
   transmit-interval {{ monitor_loop_protection.transmit_interval }}
{%     endif %}
{%     if monitor_loop_protection.disabled_time is arista.avd.defined %}
   disabled-time {{ monitor_loop_protection.disabled_time }}
{%     endif %}
```
As per EOS4.35.0F order

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,43 @@
+{%     if monitor_loop_protection.enabled is arista.avd.defined(true) %}
+| Enabled | True |
+{%     elif monitor_loop_protection.enabled is arista.avd.defined(false) %}
+| Enabled | False |
```
```suggestion
{%     if monitor_loop_protection.enabled is arista.avd.defined%}
| Enabled | {{ monitor_loop_protection.enabled }} |
```

MaheshGSLAB commented on ansible_collections/arista/avd/extensions/molecule/eos_cli_config_gen/inventory/host_vars/host1/monitor-loop-protection.yml:
```diff
@@ -0,0 +1,11 @@
+
+ethernet_interfaces:
+  - name: Ethernet 2
+    loop_protection: true
```
lets add the test in ethernet-interfaces.yml file

MaheshGSLAB commented on ansible_collections/arista/avd/extensions/molecule/eos_cli_config_gen/inventory/host_vars/host2/monitor-loop-protection.yml:
```diff
@@ -0,0 +1,11 @@
+  rate_limit: 100
+  transmit_interval: 10
+
+ethernet_interfaces:
```
same as above, add the tests in ethernet-interfaces.yml but with loop_protection: false as loop_protection: true was already tested in host1

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,43 @@
+{%         if ethernet_interface.loop_protection is arista.avd.defined(false) %}
+{{ ethernet_interface.name }}
+{%         endif %}
+{%     endfor %}
```
First we need to check if any interface has loop_protection: false then only we need to add the entry about Disabled Interfaces.
Plus give it as heading like `### Disabled Interfaces` then create a table like
| Interfaces |
| ---------- |
| Eth1 |

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,43 @@
+{% if monitor_loop_protection is arista.avd.defined %}
+
+## Monitor Loop Protestion Options
+
```
```suggestion
## Monitor Loop Protection

### Monitor Loop Protection Settings

```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2:
```diff
@@ -491,6 +491,9 @@ interface {{ ethernet_interface.name }}
 {%     if ethernet_interface.lldp.ztp_vlan is arista.avd.defined %}
    lldp tlv transmit ztp vlan {{ ethernet_interface.lldp.ztp_vlan }}
 {%     endif %}
+{%     if ethernet_interface.loop_protection is arista.avd.defined(false) %}
+   no loop-protection
+{%     endif %}
```
```suggestion
{%     if ethernet_interface.loop_protection is arista.avd.defined(false) %}
   no loop-protection
{%.    elif ethernet_interface.loop_protection is arista.avd.defined(true) %}
   loop-protection
{%     endif %}
```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,27 @@
+{#
+ Copyright (c) 2023-2025 Arista Networks, Inc.
```
```suggestion
 Copyright (c) 2025 Arista Networks, Inc.
```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,43 @@
+{#
+ Copyright (c) 2023-2025 Arista Networks, Inc.
```
```suggestion
 Copyright (c) 2025 Arista Networks, Inc.
```

dmilyt committed:
Update python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml

Co-authored-by: Mahesh Kumar <122076792+MaheshGSLAB@users.noreply.github.com>

dmilyt committed:
Update python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/monitor-loop-protection.j2

Co-authored-by: Mahesh Kumar <122076792+MaheshGSLAB@users.noreply.github.com>

dmilyt committed:
Update python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2

Co-authored-by: Mahesh Kumar <122076792+MaheshGSLAB@users.noreply.github.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dmilyt committed:
Molecule files some fixes

dmilyt committed:
fixed Copiright in *.j2 files add update docomentation template

made copiright fixes
add a loop to check disabled interfaces and crate a table for it

dmilyt commented:
/rebase

dmilyt committed:
Merge branch 'devel' into 6012

dmilyt committed:
Merge branch 'devel' into 6012

dmilyt committed:
New molucule files

dmilyt committed:
add space befor disabled intefase table

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,27 @@
+!
+monitor loop-protection
+{%     if monitor_loop_protection.enabled is arista.avd.defined(true) %}
+   no schutdown
```
```suggestion
   no shutdown
```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,47 @@
+| Enabled | {{ monitor_loop_protection.enabled }} |
+{%     endif %}
+{%     if monitor_loop_protection.disabled_time is arista.avd.defined %}
+| disabled-time | {{ monitor_loop_protection.disabled_time }} |
```
```suggestion
| Disabled Time | {{ monitor_loop_protection.disabled_time }} |
```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,47 @@
+| disabled-time | {{ monitor_loop_protection.disabled_time }} |
+{%     endif %}
+{%     if monitor_loop_protection.protect_vlan is arista.avd.defined %}
+| protect vlan | {{ monitor_loop_protection.protect_vlan }} |
```
```suggestion
| Protect Vlan | {{ monitor_loop_protection.protect_vlan }} |
```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,47 @@
+| protect vlan | {{ monitor_loop_protection.protect_vlan }} |
+{%     endif %}
+{%     if monitor_loop_protection.rate_limit is arista.avd.defined %}
+| rate-limit | {{ monitor_loop_protection.rate_limit }} |
```
```suggestion
| Rate Limit | {{ monitor_loop_protection.rate_limit }} |
```

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,47 @@
+| rate-limit | {{ monitor_loop_protection.rate_limit }} |
+{%     endif %}
+{%     if monitor_loop_protection.transmit_interval is arista.avd.defined %}
+| transmit-interval | {{ monitor_loop_protection.transmit_interval }} |
```
```suggestion
| Transmit Interval | {{ monitor_loop_protection.transmit_interval }} |
```

MaheshGSLAB commented on ansible_collections/arista/avd/extensions/molecule/eos_cli_config_gen/documentation/devices/host2.md:
```diff
@@ -808,6 +811,13 @@ load-balance cluster
+
+| Interface | Transmit | Receive |
+| --------- | -------- | ------- |
+
```
This doesn't look good, LLDP interface settings should not reflected.

MaheshGSLAB replied:
No need to address in this PR, We can fix it in another PR

MaheshGSLAB commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,47 @@
+{%         if ethernet_interface.loop_protection is arista.avd.defined(false) %}
+| {{ ethernet_interface.name }} |
+{%         endif %}
+{%     endfor %}
```
```suggestion
{%     set loop_protection_interfaces = ethernet_interfaces | arista.avd.natural_sort('name') | selectattr("loop_protection", "arista.avd.defined", false) %}
{%     if loop_protection_interfaces %}

| Disabled Interfaces |
| ------------------- |
{%     endif %}
{%     for ethernet_interface in loop_protection_interfaces %}
| {{ ethernet_interface.name }} |
{%     endfor %}
```

gmuloc committed:
Merge branch 'devel' into 6012

dmilyt committed:
Merge branch 'devel' into 6012

dmilyt commented:
Hello Mahesh,

I want to repair/reinstall  my environment, but I can not find ansible_collections/arista/avd/requirements-dev.txt any more. In GIT it is not present. Could you help me please?

Warm regards,
Dmitry


Von: Mahesh Kumar ***@***.***>
Gesendet: Dienstag, 9. Dezember 2025 09:06
An: aristanetworks/avd ***@***.***>
Cc: Lyutov, Dmitry ***@***.***>; Assign ***@***.***>
Betreff: Re: [aristanetworks/avd] Feat(eos_cli_config_gen): Add support for monitor loop-protection (PR #6226)


@MaheshGSLAB commented on this pull request.

________________________________

In python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/monitor-loop-protection.j2<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601461258>:

> @@ -0,0 +1,27 @@

+{#

+ Copyright (c) 2025 Arista Networks, Inc.

+ Use of this source code is governed by the Apache License 2.0

+ that can be found in the LICENSE file.

+#}

+{# eos - monitor loop protection #}

+{% if monitor_loop_protection is arista.avd.defined %}

+!

+monitor loop-protection

+{%     if monitor_loop_protection.enabled is arista.avd.defined(true) %}

+   no schutdown
⬇️ Suggested change

-   no schutdown

+   no shutdown

________________________________

In python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601471736>:

> + Copyright (c) 2025 Arista Networks, Inc.

+ Use of this source code is governed by the Apache License 2.0

+ that can be found in the LICENSE file.

+#}

+{# doc - monitor loop protection #}

+{% if monitor_loop_protection is arista.avd.defined %}

+

+## Monitor Loop Protection

+

+| Options | Value |

+| ------- | ----- |

+{%     if monitor_loop_protection.enabled is arista.avd.defined %}

+| Enabled | {{ monitor_loop_protection.enabled }} |

+{%     endif %}

+{%     if monitor_loop_protection.disabled_time is arista.avd.defined %}

+| disabled-time | {{ monitor_loop_protection.disabled_time }} |
⬇️ Suggested change

-| disabled-time | {{ monitor_loop_protection.disabled_time }} |

+| Disabled Time | {{ monitor_loop_protection.disabled_time }} |

________________________________

In python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601472345>:

> +#}

+{# doc - monitor loop protection #}

+{% if monitor_loop_protection is arista.avd.defined %}

+

+## Monitor Loop Protection

+

+| Options | Value |

+| ------- | ----- |

+{%     if monitor_loop_protection.enabled is arista.avd.defined %}

+| Enabled | {{ monitor_loop_protection.enabled }} |

+{%     endif %}

+{%     if monitor_loop_protection.disabled_time is arista.avd.defined %}

+| disabled-time | {{ monitor_loop_protection.disabled_time }} |

+{%     endif %}

+{%     if monitor_loop_protection.protect_vlan is arista.avd.defined %}

+| protect vlan | {{ monitor_loop_protection.protect_vlan }} |
⬇️ Suggested change

-| protect vlan | {{ monitor_loop_protection.protect_vlan }} |

+| Protect Vlan | {{ monitor_loop_protection.protect_vlan }} |

________________________________

In python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601472937>:

> +

+## Monitor Loop Protection

+

+| Options | Value |

+| ------- | ----- |

+{%     if monitor_loop_protection.enabled is arista.avd.defined %}

+| Enabled | {{ monitor_loop_protection.enabled }} |

+{%     endif %}

+{%     if monitor_loop_protection.disabled_time is arista.avd.defined %}

+| disabled-time | {{ monitor_loop_protection.disabled_time }} |

+{%     endif %}

+{%     if monitor_loop_protection.protect_vlan is arista.avd.defined %}

+| protect vlan | {{ monitor_loop_protection.protect_vlan }} |

+{%     endif %}

+{%     if monitor_loop_protection.rate_limit is arista.avd.defined %}

+| rate-limit | {{ monitor_loop_protection.rate_limit }} |
⬇️ Suggested change

-| rate-limit | {{ monitor_loop_protection.rate_limit }} |

+| Rate Limit | {{ monitor_loop_protection.rate_limit }} |

________________________________

In python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601473523>:

> +| Options | Value |

+| ------- | ----- |

+{%     if monitor_loop_protection.enabled is arista.avd.defined %}

+| Enabled | {{ monitor_loop_protection.enabled }} |

+{%     endif %}

+{%     if monitor_loop_protection.disabled_time is arista.avd.defined %}

+| disabled-time | {{ monitor_loop_protection.disabled_time }} |

+{%     endif %}

+{%     if monitor_loop_protection.protect_vlan is arista.avd.defined %}

+| protect vlan | {{ monitor_loop_protection.protect_vlan }} |

+{%     endif %}

+{%     if monitor_loop_protection.rate_limit is arista.avd.defined %}

+| rate-limit | {{ monitor_loop_protection.rate_limit }} |

+{%     endif %}

+{%     if monitor_loop_protection.transmit_interval is arista.avd.defined %}

+| transmit-interval | {{ monitor_loop_protection.transmit_interval }} |
⬇️ Suggested change

-| transmit-interval | {{ monitor_loop_protection.transmit_interval }} |

+| Transmit Interval | {{ monitor_loop_protection.transmit_interval }} |

________________________________

In ansible_collections/arista/avd/extensions/molecule/eos_cli_config_gen/documentation/devices/host2.md<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601485591>:

> +#### LLDP Interface Settings

+

+LLDP is **disabled** globally. Local interface configs will not apply.

+

+| Interface | Transmit | Receive |

+| --------- | -------- | ------- |

+

This doesn't look good, LLDP interface settings should not reflected.

________________________________

In python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601522093>:

> +

+{%     for ethernet_interface in ethernet_interfaces | arista.avd.natural_sort('name') %}

+{%         if ethernet_interface.loop_protection is arista.avd.defined(false) %}

+| Disabled Interfaces |

+| ------------------- |

+{%             break %}

+{%         endif %}

+{%     endfor %}

+{%     for ethernet_interface in ethernet_interfaces | arista.avd.natural_sort('name') %}

+{%         if ethernet_interface.loop_protection is arista.avd.defined(false) %}

+| {{ ethernet_interface.name }} |

+{%         endif %}

+{%     endfor %}
⬇️ Suggested change

-

-{%     for ethernet_interface in ethernet_interfaces | arista.avd.natural_sort('name') %}

-{%         if ethernet_interface.loop_protection is arista.avd.defined(false) %}

-| Disabled Interfaces |

-| ------------------- |

-{%             break %}

-{%         endif %}

-{%     endfor %}

-{%     for ethernet_interface in ethernet_interfaces | arista.avd.natural_sort('name') %}

-{%         if ethernet_interface.loop_protection is arista.avd.defined(false) %}

-| {{ ethernet_interface.name }} |

-{%         endif %}

-{%     endfor %}

+{%     set loop_protection_interfaces = ethernet_interfaces | arista.avd.natural_sort('name') | selectattr("loop_protection", "arista.avd.defined", false) %}

+{%     if loop_protection_interfaces %}

+

+| Disabled Interfaces |

+| ------------------- |

+{%     endif %}

+{%     for ethernet_interface in loop_protection_interfaces %}

+| {{ ethernet_interface.name }} |

+{%     endfor %}

________________________________

In ansible_collections/arista/avd/extensions/molecule/eos_cli_config_gen/documentation/devices/host2.md<https://github.com/aristanetworks/avd/pull/6226#discussion_r2601488641>:

> +#### LLDP Interface Settings

+

+LLDP is **disabled** globally. Local interface configs will not apply.

+

+| Interface | Transmit | Receive |

+| --------- | -------- | ------- |

+

No need to address in this PR, We can fix it in another PR

—
Reply to this email directly, view it on GitHub<https://github.com/aristanetworks/avd/pull/6226#pullrequestreview-3555932928>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARURW4FQYHSCCAIJAJWHR5L4AZ7E7AVCNFSM6AAAAACOEIOQ5CVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTKNJVHEZTEOJSHA>.
You are receiving this because you were assigned.Message ID: ***@***.***>

Controlware GmbH | Telefon: +49 6074 858-00 | Fax: +49 6074 858-108 | E-Mail: ***@***.*** | https://www.controlware.de | Sitz: 63128 Dietzenbach | Registergericht: Offenbach a.M., HRB Nr. 6431 | Geschäftsführer: Bernd Schwefing, Michael Küchen | Aufsichtsratsvorsitzender: Christof Ziegler

dmilyt committed:
Fixed requested typo errors. And inprove some code in documentation

dmilyt committed:
new molecule files

dmilyt committed:
Merge branch 'devel' into 6012

Shivani-gslab commented on python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/monitor_loop_protection.yml:
```diff
@@ -0,0 +1,41 @@
+        description: |-
+          VLAN range as string.
+          "< vlan_id >, < vlan_id >-< vlan_id >"#
+          Example: 15,16,17,18
```
```suggestion
          "< vlan_id >, < vlan_id >-< vlan_id >"
          Example: 15,16,17,18
```

Shivani-gslab commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,43 @@
+{%     endif %}
+{%     for ethernet_interface in loop_protection_interfaces %}
+| {{ ethernet_interface.name }} |
+{%     endfor %}
```
```suggestion
{%     if loop_protection_interfaces %}

| Disabled Interfaces |
| ------------------- |
{%         for ethernet_interface in loop_protection_interfaces %}
| {{ ethernet_interface.name }} |
{%         endfor %}
{%     endif %}
```

gmuloc committed:
Update python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/monitor_loop_protection.yml

Co-authored-by: Shivani-gslab <145646625+Shivani-gslab@users.noreply.github.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dmilyt committed:
Merge branch 'devel' into 6012

gmuloc committed:
Update python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2

Co-authored-by: Shivani-gslab <145646625+Shivani-gslab@users.noreply.github.com>

gmuloc committed:
Merge branch 'devel' into 6012

Shivani-gslab reviewed (approved):
LGTM

gmuloc commented on python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/monitor_loop_protection.yml:
```diff
@@ -0,0 +1,41 @@
+        type: int
+        convert_types:
+          - str
+        description: Loop protection packet transmit interval. Default is 5 seconds.
```
```suggestion
        description: Loop protection packet transmit interval. EOS default is 5 seconds.
```

gmuloc commented on python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/monitor_loop_protection.yml:
```diff
@@ -0,0 +1,41 @@
+        type: int
+        convert_types:
+          - str
+        description: Rate limits the loop detection frames. Default to maximum 1000/second.
```
```suggestion
        description: Rate limits the loop detection frames. EOS default is 1000/second.
```

gmuloc commented on python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/monitor_loop_protection.yml:
```diff
@@ -0,0 +1,41 @@
+        convert_types:
+          - str
+        description: |-
+          Port disable time. Default is 604800 seconds (7 days).
```
```suggestion
          Port disable time. EOS default is 604800 seconds (7 days).
```

gmuloc commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,43 @@
+{%     if monitor_loop_protection.transmit_interval is arista.avd.defined %}
+| Transmit-interval | {{ monitor_loop_protection.transmit_interval }} |
+{%     endif %}
+{%     set loop_protection_interfaces = ethernet_interfaces | arista.avd.natural_sort('name') | selectattr("loop_protection", "arista.avd.defined", false) %}
```
```suggestion
{%     set loop_protection_interfaces = ethernet_interfaces | selectattr("loop_protection", "arista.avd.defined", false) | arista.avd.natural_sort('name') %}
```

gmuloc commented on ansible_collections/arista/avd/extensions/molecule/eos_cli_config_gen/inventory/host_vars/host1/ethernet-interfaces.yml:
```diff
@@ -2133,3 +2134,8 @@ ethernet_interfaces:
       transmitter:
         signal_power: -20
+
+# Monitor loop protection
```
```suggestion
  # Disable monitor loop protection
```

gmuloc commented on python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/monitor-loop-protection.j2:
```diff
@@ -0,0 +1,43 @@
+{%     set loop_protection_interfaces = ethernet_interfaces | arista.avd.natural_sort('name') | selectattr("loop_protection", "arista.avd.defined", false) %}
+{%     if loop_protection_interfaces %}
+
+| Disabled Interfaces |
```
does it need to be a separate table? Could it be another row in the table above?

Also question looking at the tests - what does it mean to have interfaces disabled for loop protection when loop-protection is disabled?

dmilyt replied:
A separate table was requested by Mahesh Kumar MaheshGSLAB  https://github.com/aristanetworks/avd/pull/6226#discussion_r2592490180 Schell I insert it in the table above as separate column?

gmuloc replied:
Hi @dmilyt I have edited the doc to align with how we are doing things today.
Also found an issue when ethernet_interfaces is not defiend and just fixed it. I will merge after a final review.

Thanks a lot for your work!

gmuloc committed:
Apply suggestions from code review

gmuloc committed:
Update ansible_collections/arista/avd/extensions/molecule/eos_cli_config_gen/inventory/host_vars/host1/ethernet-interfaces.yml

dmilyt committed:
Merge branch 'devel' into 6012

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gmuloc committed:
Refactor: Edit documentation template

dmilyt committed:
Merge branch 'devel' into 6012

gmuloc committed:
Test: Moleculing some molecule

gmuloc committed:
Fix: Template erroring out when no ethernet interface is defined

gmuloc reviewed (approved):
LGTM if CI passes. Thanks @dmilyt

dmilyt committed:
Merge branch 'devel' into 6012

gmuloc committed:
Merge branch 'devel' into 6012
